amethyst_rendy/shaders/fragment/header/ltc.glsl

```diff
@@ -0,0 +1,540 @@
+#ifndef LTC_FRAG
+#define LTC_FRAG
+
+// Get uv coordinates into LTC lookup texture
+//
+// N: Normal
+// V: View direction
+// roughness
+vec2 ltc1_coords(float NdotV, float roughness) {
+    vec2 coords = vec2(roughness, sqrt(1.0 - saturate(NdotV)));
+
+    /* Scale and bias coordinates, for correct filtered lookup */
+    coords = coords*LUT_SCALE + LUT_BIAS;
+
+    return coords;
+}
+// Get uv coords into LTC lookup tabled of precomputed clipped form factors
+//
+// dir: F direction
+// len: form factor of the polygon
+vec2 ltc2_coords(float dir, float len) {
+    vec2 coords = vec2(dir*0.5 + 0.5, len);
+
+    /* Scale and bias coordinates, for correct filtered lookup */
+    coords = coords*LUT_SCALE + LUT_BIAS;
+
+    return coords;
+}
+
+/** Get inverse matrix from LTC lookup texture */
+mat3 ltc_matrix(sampler2D tex, vec2 coord) {
+    const vec4 t = texture(tex, coord);
+    mat3 Minv = mat3(
+        vec3(  t.x,   0, t.y),
+        vec3(  0, 1,   0),
+        vec3(t.z,   0, t.w)
+    );
+
+    return Minv;
+}
+// An extended version of the implementation from
+// "How to solve a cubic equation, revisited"
+// http://momentsingraphics.de/?p=105
+vec3 solve_cubic(vec4 Coefficient)
+{
+    // Normalize the polynomial
+    Coefficient.xyz /= Coefficient.w;
+    // Divide middle coefficients by three
+    Coefficient.yz /= 3.0;
+
+    float A = Coefficient.w;
```

Comment:
Can we get all these variables in the shader to match a uniform scheme? The varying capitalization and spacing and camel casing and snake casing makes my head hurt.

Reply:
Definitely. I asked in discord about coding guidlines for shaders but we don't have any atm.
This will change when I do my documentation pass for the shaders. Before that I wanted some structural feedback.